sql/src/test/java/org/opensearch/sql/sql/parser/AstBuilderTest.java

```diff
@@ -86,6 +87,19 @@ public void can_build_select_all_from_index() {
     assertThrows(SyntaxCheckException.class, () -> buildAST("SELECT *"));
   }
 
+  @Test
+  public void can_build_nested_select_all() {
+    assertEquals(
+        project(
+            relation("test"),
+            alias("nested(field.*)",
+                new NestedAllTupleFields("field")
+            )
+        ),
+        buildAST("SELECT nested(field.*) FROM test")
```

Comment:
Is there any way to use nested(field.*) in another clause?  It won't give a syntax error, but maybe it will fail later on. 
Consider this...

Reply:
Yes it is only valid in the `SELECT` clause but a user could use it in another clause and cause a nullPointerException. I can add some safeguards.